plugins/modules/homebrew.py

```diff
@@ -405,15 +406,23 @@ def _extract_package_name(self, package_detail, is_cask):
         #
         # Issue https://github.com/ansible-collections/community.general/issues/10012:
         # package_detail["tap"] is None if package is no longer available.
-        tapped_name = [package_detail["tap"] + "/" + name] if package_detail["tap"] else []
-        aliases = package_detail.get("aliases", [])
-        package_names = set([name, full_name] + tapped_name + aliases)
+        # 
+        # Issue https://github.com/ansible-collections/community.general/issues/10804
+        # name can be an alias, oldnames or old_tokens optionally prefixed by tap
+        package_names = { name, full_name }
+        package_names.update(package_detail.get("aliases", []))
+        package_names.update(package_detail.get("oldnames", []))
+        package_names.update(package_detail.get("old_tokens", []))
+        if package_detail['tap']:
+            # names so far, with tap prefix added to each
+            tapped_names = {f"{package_detail["tap"]}/{x}" for x in package_names}
+            package_names.update(tapped_names)
 
         # Finally, identify which of all those package names was the one supplied by the user.
         package_names = package_names & set(self.packages)
         if len(package_names) != 1:
             self.failed = True
-            self.message = "Package names are missing or ambiguous: " + ", ".join(str(p) for p in package_names)
+            self.message = f"Package names for {name} are missing or ambiguous: {', '.join(str(p) for p in package_names)}"
```

Comment:
Please do not use f-strings. They do not work on Python 2. (You can use `str.format()`.)